src/words/WordsToReview.js

```diff
@@ -11,39 +18,178 @@ export default function WordsToReview({
   notifyWordChanged,
   source,
 }) {
+  const totalWordsTranslated = words.length;
+  const [inEditMode, setInEditMode] = useState(false);
+  const [wordsForExercises, setWordsForExercises] = useState([]);
+  const [wordsExcludedForExercises, setWordsExcludedForExercises] = useState(
+    [],
+  );
+  const [totalWordsSelectedByZeeguu, setTotalWordsSelectedByZeeguu] =
+    useState();
+  const [totalWordsEditedByUser, setTotalWordsEditedByUser] = useState();
+  const [wordsExpressions, setWordsExpressions] = useState([]);
+  const [showExplainWordSelectionModal, setShowExplainWordSelectionModal] =
+    useState(false);
+
+  useEffect(() => {
+    let newWordsForExercises = [];
+    let newWordsExcludedExercises = [];
+    let newWordExpressions = [];
+    // Keep track how many words Zeeguu selected
+    let newTotalWordsSelectedByZeeguu = 0;
+    let newTotalWordsEditedByUser = 0;
+
+    for (let i = 0; i < words.length; i++) {
+      let word = words[i];
+      if (
+        word.fit_for_study &&
+        word.user_preference !== USER_WORD_PREFERENCE.USE_IN_EXERCISES
+      )
+        newTotalWordsSelectedByZeeguu += 1;
+      if (word.user_preference !== USER_WORD_PREFERENCE.NO_PREFERENCE)
+        newTotalWordsEditedByUser += 1;
+      if (tokenize(word.from).length >= 3) newWordExpressions.push(word);
+      else if (!word.fit_for_study) newWordsExcludedExercises.push(word);
+      else newWordsForExercises.push(word);
+    }
+    setWordsForExercises(newWordsForExercises);
+    setWordsExcludedForExercises(newWordsExcludedExercises);
+    setWordsExpressions(newWordExpressions);
+    setTotalWordsSelectedByZeeguu(newTotalWordsSelectedByZeeguu);
+    setTotalWordsEditedByUser(newTotalWordsEditedByUser);
+  }, [words]);
+
+  if (words.length === 0)
+    return (
+      <>
+        {" "}
+        <div style={{ marginTop: "5em" }}>
+          <h1>{strings.ReviewTranslations}</h1>
+          <p>
+            <b>{strings.from}</b>
+            {articleInfo.title}
+          </p>
+        </div>
+        <Infobox>
+          <div>
+            <p>You didn't translate any words in this article.</p>
+          </div>
+        </Infobox>
+      </>
+    );
+
+  const isNoWordsSelected = wordsForExercises.length === 0;
+  const isZeeguuSelectWords =
```

Comment:
i know it's a predicate, but sounds bad in English. can we rename to `hasZeeguuSelectedWords` maybe?